### PR TITLE
Switch audio devices

### PR DESCRIPTION
**Pull request by mathias5r**
Adding support of switching between multiple audio devices (phone, speaker, bluetooth and headset).

### CODE REVIEW

File: android/src/main/java/io/wazo/callkeep/RNCallKeepModule.java [hunk 1]
```diff
@@ -491,6 +493,76 @@ public void toggleAudioRouteSpeaker(String uuid, boolean routeSpeaker) {
...
+    @ReactMethod
+    public void setAudioRoute(String uuid, String audioRoute, Promise promise){
+        try {
+            VoiceConnection conn = (VoiceConnection) VoiceConnectionService.getConnection(uuid);
+            if (conn == null) {
+                return;
+            }
+            if(audioRoute.equals("Bluetooth")) {
+                conn.setAudioRoute(CallAudioState.ROUTE_BLUETOOTH);
+                promise.resolve("Successfully set bluetooth route");
```
Comment (manuquentin) [inline android/src/main/java/io/wazo/callkeep/RNCallKeepModule.java:505]: Should we just resolve a boolean here to simplify the result check ?

File: android/src/main/java/io/wazo/callkeep/RNCallKeepModule.java [hunk 2]
```diff
@@ -491,6 +493,76 @@ public void toggleAudioRouteSpeaker(String uuid, boolean routeSpeaker) {
...
 
+    @ReactMethod
+    public void setAudioRoute(String uuid, String audioRoute, Promise promise){
+        try {
+            VoiceConnection conn = (VoiceConnection) VoiceConnectionService.getConnection(uuid);
+            if (conn == null) {
+                return;
+            }
+            if(audioRoute.equals("Bluetooth")) {
+                conn.setAudioRoute(CallAudioState.ROUTE_BLUETOOTH);
```
Comment (manuquentin) [inline android/src/main/java/io/wazo/callkeep/RNCallKeepModule.java:504]: Can you add logs (eg: `Log.d(Tag, ...)`) about the action and the current audioRoute at this point ?

**Review by manuquentin — CHANGES_REQUESTED** [review]

File: android/src/main/java/io/wazo/callkeep/RNCallKeepModule.java (hunk 1 above)
Comment (mathias5r) [reply android/src/main/java/io/wazo/callkeep/RNCallKeepModule.java]: Done!

File: android/src/main/java/io/wazo/callkeep/RNCallKeepModule.java (hunk 2 above)
Comment (mathias5r) [reply android/src/main/java/io/wazo/callkeep/RNCallKeepModule.java:505]: Done!

**Review by manuquentin — APPROVED** [review]

**Comment by manuquentin** [comment]
Thanks @mathias5r !

**Comment by rcidt** [comment]
Does this support selecting from multiple connected devices?

**Comment by AbdulBsit** [comment]
@mathias5r @manuquentin  Hey, i have a question regarding activating audio session in [getAudioInputs](https://github.com/react-native-webrtc/react-native-callkeep/blob/a0d5c334f590b25fe67d5c7923890679040a414e/ios/RNCallKeep/RNCallKeep.m#L613)  is there any specific reason, why we re-activated session when we have already done from `didActivateAudioSession`, does getting input devices requires this step 
i.e., 
- setting category again in case it's not there and 
- setActive to true, 

Also in case we remove these step, what are the possible flaws expected in any part of lifecycle of call

**Pull request merged**